### PR TITLE
LAVA DWARVES 2: Strike the Earth!

### PR DESCRIPTION
**Pull request by Iamgoofball**
# Read the wiki page here https://tgstation13.org/wiki/Lava_Dwarf

Get ready to dorf. Icons fixed and everything.

### CODE REVIEW

**Comment by GunHog** [comment]
Good to see this back! I must prepare my milk sword!

**Comment by ShizCalev** [comment]
why aren't they tgm

File: code/_onclick/hud/hud.dm [hunk 1]
```diff
@@ -21,6 +21,8 @@
 	var/obj/screen/alien_plasma_display
 	var/obj/screen/alien_queen_finder
 
+	var/obj/screen/alcohol_amount
```
Comment (Really-Good-Soda-Flavor) [inline code/_onclick/hud/hud.dm]: Unused var, get rid of it.

File: code/_onclick/hud/hud.dm [hunk 2]
```diff
@@ -112,6 +114,7 @@
 	blobpwrdisplay = null
 	alien_plasma_display = null
 	alien_queen_finder = null
+	alcohol_amount = null
```
Comment (Really-Good-Soda-Flavor) [inline code/_onclick/hud/hud.dm]: Unused var.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Really-Good-Soda-Flavor** [comment]
Yea convert the maps to TGM if they aren't

File: _maps/RandomRuins/LavaRuins/lavaland_surface_dorffort.dmm [hunk 3]
```diff
@@ -0,0 +1,55 @@
+"a" = (/turf/template_noop,/area/template_noop)
```
Comment (Cyberboss) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_dorffort.dmm]: TGM

File: _maps/RandomRuins/LavaRuins/lavaland_surface_dorfwagon.dmm [hunk 4]
```diff
@@ -0,0 +1,45 @@
+"a" = (/turf/template_noop,/area/template_noop)
```
Comment (Cyberboss) [inline _maps/RandomRuins/LavaRuins/lavaland_surface_dorfwagon.dmm]: TGM

File: _maps/templates/dwarf/brewery.dmm [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"a" = (/turf/closed/wall,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/brewery.dmm]: TGM

File: _maps/templates/dwarf/dining_hall.dmm [hunk 6]
```diff
@@ -0,0 +1,19 @@
+"a" = (/turf/closed/wall,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/dining_hall.dmm]: TGM

File: _maps/templates/dwarf/dorm.dmm [hunk 7]
```diff
@@ -0,0 +1,13 @@
+"a" = (/turf/closed/wall,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/dorm.dmm]: TGM

File: _maps/templates/dwarf/farm.dmm [hunk 8]
```diff
@@ -0,0 +1,22 @@
+"a" = (/turf/closed/wall,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/farm.dmm]: TGM

File: _maps/templates/dwarf/smeltery.dmm [hunk 9]
```diff
@@ -0,0 +1,15 @@
+"a" = (/turf/closed/wall,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/smeltery.dmm]: TGM

File: _maps/templates/dwarf/throne_room.dmm [hunk 10]
```diff
@@ -0,0 +1,22 @@
+"a" = (/turf/closed/wall/mineral/gold,/area/awaymission)
```
Comment (Cyberboss) [inline _maps/templates/dwarf/throne_room.dmm]: TGM

File: code/_onclick/hud/hud.dm (hunk 1 above)
Comment (Cyberboss) [inline code/_onclick/hud/hud.dm]: Unused var

File: code/_onclick/hud/hud.dm (hunk 2 above)
Comment (Cyberboss) [inline code/_onclick/hud/hud.dm]: Unused var

File: code/_onclick/item_attack.dm [hunk 11]
```diff
@@ -50,7 +50,15 @@
...
 	add_fingerprint(user)
-
+	if(smelted_material)
+		smelted_material.reaction_mob(M, TOUCH, 30)
+		take_damage(smelted_material.durability_reduction, BRUTE, "melee", 1)
+
+/obj/item/obj_break(damage_flag)
+	var/turf/T = get_turf(src)
+	T.visible_message("<span class = 'danger'>[src] breaks apart!</span>")
+	qdel(src)
```
Comment (Cyberboss) [inline code/_onclick/item_attack.dm]: Why? this totally negates the point of obj damage!

File: code/game/gamemodes/wizard/artefact.dm [hunk 12]
```diff
@@ -22,32 +22,34 @@
...
-	src.spawn_path = spawn_type
-	src.spawn_amt_left = spawn_amt
-	src.desc = desc
-	src.spawn_fast = spawn_fast
+/obj/structure/rend/hfs
+	spawn_path = /mob/living/simple_animal/hostile/faithless
+	spawn_amt_left = 30
+
+/obj/structure/rend/New()
 	START_PROCESSING(SSobj, src)
```
Comment (Cyberboss) [inline code/game/gamemodes/wizard/artefact.dm]: Initialize() and . = ..()

File: code/game/objects/items.dm [hunk 13]
```diff
@@ -106,6 +106,9 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 	var/datum/rpg_loot/rpg_loot = null
 
+	var/datum/reagent/smelted_material = null
+	var/starting_material = null
```
Comment (Cyberboss) [inline code/game/objects/items.dm]: Remove the ` = null`s

File: code/game/turfs/simulated/minerals.dm [hunk 14]
```diff
@@ -141,7 +144,7 @@
 
 /turf/closed/mineral/random
 	var/mineralSpawnChanceList
-		//Currently, Adamantine won't spawn as it has no uses. -Durandan
+		//Currently, Adamantine won't spawn as it has no uses. -Durandan // HAHA FUCK YOU IT DOES NOW BITCHES
```
Comment (Cyberboss) [inline code/game/turfs/simulated/minerals.dm]: Just remove the comment

File: code/game/turfs/simulated/minerals.dm [hunk 15]
```diff
@@ -377,6 +384,37 @@
...
+	mineralType = /obj/item/weapon/ore/adamantine
+	mineralAmt = 1
+	spreadChance = 0
+	spread = 0
+	scan_state = "rock_adamantine"
+	environment_type = "basalt"
+	turf_type = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
+	baseturf = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
+	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	defer_change = 1
```
Comment (Cyberboss) [inline code/game/turfs/simulated/minerals.dm]: TRUE

File: code/game/turfs/simulated/minerals.dm [hunk 16]
```diff
@@ -377,6 +384,37 @@
 	turf_type = /turf/open/floor/plating/ashplanet/rocky
 	defer_change = 1
 
+/turf/closed/mineral/adamantine
+	mineralType = /obj/item/weapon/ore/adamantine
+	mineralAmt = 1
+	spreadChance = 0
+	spread = 0
```
Comment (Cyberboss) [inline code/game/turfs/simulated/minerals.dm]: FALSE

File: code/game/turfs/simulated/minerals.dm [hunk 17]
```diff
@@ -377,6 +384,37 @@
...
+		GLOB.secret_triggered = TRUE
+		visible_message("<span class = 'userdanger'>You uncover an ancient evil!</span>")
+		new /obj/structure/rend/hfs(src)
+	..()
+
+
+/turf/closed/mineral/clay
+	mineralType = /obj/item/stack/sheet/mineral/clay
+	spreadChance = 20
+	spread = 1
```
Comment (Cyberboss) [inline code/game/turfs/simulated/minerals.dm]: TRUE

File: code/game/turfs/simulated/minerals.dm [hunk 18]
```diff
@@ -377,6 +384,37 @@
...
+	scan_state = "rock_clay"
+	environment_type = "basalt"
+	turf_type = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
+	baseturf = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
+	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	defer_change = 1
```
Comment (Cyberboss) [inline code/game/turfs/simulated/minerals.dm]: TRUE

File: code/modules/clothing/dwarf_clothes.dm [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+	desc = "Protects the head from tantrums."
+	icon_state = "dwarf"
+	item_state = "dwarf"
+	species_clothing_whitelist = list("dwarf")
+
+/obj/item/clothing/head/helmet/dwarf/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/helmet_plating/S = locate() in contents
+	if(S)
+		var/image/Q = image(icon, icon_state)
```
Comment (Cyberboss) [inline code/modules/clothing/dwarf_clothes.dm]: Use mutable_appearance

File: code/modules/clothing/dwarf_clothes.dm [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+	heat_protection = CHEST|GROIN|ARMS
+	armor = list(melee = 50, bullet = 10, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 80, acid = 80)
+	strip_delay = 80
+	species_clothing_whitelist = list("dwarf")
+
+/obj/item/clothing/suit/armor/vest/dwarf/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/armor_plating/S = locate() in contents
+	if(S)
+		var/image/Q = image(icon, icon_state)
```
Comment (Cyberboss) [inline code/modules/clothing/dwarf_clothes.dm]: Use mutable_appearance

File: code/game/objects/structures/holosign.dm [hunk 21]
```diff
@@ -128,3 +128,15 @@
...
 			shockcd = TRUE
 			addtimer(CALLBACK(src, .proc/cooldown), 5)
+
+/obj/structure/holosign/mandate
+	name = "Royal Mandate"
+	desc = "The mandate reads:."
+	icon_state = "mandate"
+	var/mandate = "eat shit and die"
+	obj_integrity = INFINITY // Remove with the scepter itself
+	max_integrity = INFINITY
```
Comment (Cyberboss) [inline code/game/objects/structures/holosign.dm]: No, just give it INDESTRUCTIBLE

File: code/game/objects/structures/holosign.dm [hunk 22]
```diff
@@ -128,3 +128,15 @@
 			M.electrocute_act(15,"Energy Barrier", safety=1)
 			shockcd = TRUE
 			addtimer(CALLBACK(src, .proc/cooldown), 5)
+
+/obj/structure/holosign/mandate
+	name = "Royal Mandate"
+	desc = "The mandate reads:."
+	icon_state = "mandate"
+	var/mandate = "eat shit and die"
```
Comment (Cyberboss) [inline code/game/objects/structures/holosign.dm]: New vars after parent vars

File: code/modules/food_and_drinks/drinks/brewing_barrel.dm [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+			if(A["booze_power"] >= current_highest_boozepwr["power"])
+				current_highest_boozepwr["prefix"] = A["prefix"]
+				current_highest_boozepwr["power"] = A["booze_power"]
+		var/final_name = current_highest_boozepwr["prefix"]
+		if(current_highest_boozepwr["prefix"] == "plump-helmet")
+			AB.name = "dwarven wine"
+		else
+			AB.name = "[final_name] ale"
+		brewables = list()
+		already_brewing = list()
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/drinks/brewing_barrel.dm]: .Cut() for both of these

File: code/modules/mining/lavaland/ruins/dwarfruin.dm [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+		return
+	if(spawns_left)
+		spawns_left--
+		var/client/new_dorf = user.client
+		spawn_dorf(new_dorf)
+	else
+		to_chat(user, "There's no more room at the fortress for new migrants! Wait for them to build a new dormitory.")
+
+/obj/machinery/migrant_spawner/proc/spawn_dorf(client/new_dorf)
+	var/mob/living/carbon/human/M = new/mob/living/carbon/human(get_turf(src))
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/ruins/dwarfruin.dm]: `var/mob/living/carbon/human/M = new (get_turf(src))`

File: code/modules/mining/lavaland/ruins/dwarfruin.dm [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+	M.set_species(/datum/species/dwarf)
+	M.key = new_dorf.key
+	M.equipOutfit(dorf_gear)
+	M.real_name = M.dna.species.random_name()
+	to_chat(M, "<span class='notice'>You have arrived. After a journey from The Mountainhomes into the forbidding beyond.\
+	 Your harsh trek has finally ended. A new chapter of Dwarven history begins at this place 'Koganusan' -Boatmurdered- Strike the Earth.<br><br>There's been rumors of a station occupied by Humans nearby. It's dangerous to fight them, so you probably shouldn't invade unless they really provoke you.<br>The rest of lavaland is rightfully the Dwarven peoples', and you should forge your own path for the empire, be it friendly or hostile.</span>")
+	M.mind.special_role = "Dwarf"
+	var/datum/objective/O = new("Make claim to these hellish lands, constructing a great fortress and mining the rock for its precious metals, in the name of the Dwarven people!")
+	M.mind.objectives += O
+	LAZYADD(GLOB.dwarves_list, M)
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/ruins/dwarfruin.dm]: Where are they removed from the list?

File: code/modules/mining/lavaland/ruins/dwarfruin.dm [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+	to_chat(M, "<b>Be sure to read the wiki page at https://tgstation13.org/wiki/Lava_Dwarf to learn how to strike the earth!</b>")
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/ruins/dwarfruin.dm]: ock ick

File: code/modules/mining/lavaland/ruins/dwarfruin.dm [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+	back = /obj/item/weapon/storage/backpack/satchel/leather
+
+/*Misc Items + Crafting Parts*/
+/obj/item/weapon/reagent_containers/food/drinks/wooden_mug
+	name = "wooden mug"
+	desc = "A mug for serving hearty brews."
+	icon = 'icons/obj/drinks.dmi'
+	item_state = "manlydorfglass"
+	icon_state = "manlydorfglass"
+	spillable = 1
```
Comment (Cyberboss) [inline code/modules/mining/lavaland/ruins/dwarfruin.dm]: TRUE

File: code/modules/mining/ores_coins.dm [hunk 28]
```diff
@@ -11,6 +11,11 @@
 	icon_state = "ore"
 	var/points = 0 //How many points this ore gets you from the ore redemption machine
 	var/refined_type = null //What this ore defaults to being refined into
+	var/reagent_id = null
```
Comment (Cyberboss) [inline code/modules/mining/ores_coins.dm]: remove ` = null`

File: code/modules/mining/smelter.dm [hunk 29]
```diff
@@ -0,0 +1,331 @@
...
+/obj/machinery/anvil/attackby(obj/item/weapon/W, mob/user, params)
+	if(!isdwarf(user))
+		to_chat(user, "You don't comprehend this tool well enough to use it.")
+		return
+	if(!istype(W, /obj/item/weapon/smith_hammer))
+		return ..()
+	if(!current_mold && istype(W, /obj/item/weapon/reagent_containers/glass/mold))
+		var/obj/item/weapon/reagent_containers/glass/mold/M = W
+		var/datum/reagent/R = M.reagents.get_master_reagent()
+		if(R && R.volume == 25)
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: Magic number

File: code/modules/mining/smelter.dm [hunk 30]
```diff
@@ -0,0 +1,331 @@
...
+	if(istype(W, /obj/item/weapon/smith_hammer))
+		if(current_mold)
+			to_chat(user, "You break the result out of the mold.")
+			new current_mold.type(get_turf(src))
+			var/datum/reagent/R = current_mold.reagents.get_master_reagent()
+			if(!R)
+				QDEL_NULL(current_mold) //not sure what this does but it apparently deletes null stuff, go figure
+				cut_overlay(my_mold)
+				my_mold = null
+				current_mold = null
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: The QDEL_NULL already does this

File: code/modules/mining/smelter.dm [hunk 31]
```diff
@@ -0,0 +1,331 @@
...
+				I.smelted_material = new R.type()
+				I.post_smithing()
+			else
+				I = new R.produce_type(get_turf(src))
+				var/obj/item/stack/S = I
+				S.amount = 5
+			QDEL_NULL(current_mold)
+			cut_overlay(my_mold)
+			my_mold = null
+			current_mold = null
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: The QDEL_NULL already does this

File: code/modules/mining/smelter.dm [hunk 32]
```diff
@@ -0,0 +1,331 @@
...
+				I = new current_mold.produce_type(get_turf(src))
+				I.smelted_material = new R.type()
+				I.post_smithing()
+			else
+				I = new R.produce_type(get_turf(src))
+				var/obj/item/stack/S = I
+				S.amount = 5
+			QDEL_NULL(current_mold)
+			cut_overlay(my_mold)
+			my_mold = null
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: QDEL_NULL

File: code/modules/mining/smelter.dm [hunk 33]
```diff
@@ -0,0 +1,331 @@
...
+			return FALSE
+	if(istype(W, /obj/item/weapon/smith_hammer))
+		if(current_mold)
+			to_chat(user, "You break the result out of the mold.")
+			new current_mold.type(get_turf(src))
+			var/datum/reagent/R = current_mold.reagents.get_master_reagent()
+			if(!R)
+				QDEL_NULL(current_mold) //not sure what this does but it apparently deletes null stuff, go figure
+				cut_overlay(my_mold)
+				my_mold = null
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: QDEL_NULL

File: code/modules/mining/smelter.dm [hunk 34]
```diff
@@ -0,0 +1,331 @@
...
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "claymore"
+	item_state = "claymore"
+
+/obj/item/weapon/smithed_sword/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/blade/B = locate() in contents
+	if(B)
+		var/image/I = image('icons/obj/blacksmithing.dmi', "sword_blade")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mining/smelter.dm [hunk 35]
```diff
@@ -0,0 +1,331 @@
...
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = SLOT_BACK
+	throw_speed = 4
+	attack_verb = list("attacked", "poked", "jabbed", "torn", "gored")
+
+/obj/item/weapon/twohanded/smithed_warhammer/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/hammer_head/M = locate() in contents
+	if(M)
+		var/image/I = image('icons/obj/blacksmithing.dmi', "hammer_head")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mining/smelter.dm [hunk 36]
```diff
@@ -0,0 +1,331 @@
...
+	desc = "A pickaxe made of unobtanium, you probably shouldn't be seeing this."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "spickaxe"
+	item_state = "spickaxe"
+
+/obj/item/weapon/pickaxe/smithed_pickaxe/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/pickaxe_head/P = locate() in contents
+	if(P)
+		var/image/O = image('icons/obj/blacksmithing.dmi', "pickaxe_head")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mining/smelter.dm [hunk 37]
```diff
@@ -0,0 +1,331 @@
...
+	name = "unobtanium shovel"
+	desc = "A shovel made of unobtanium, you probably shouldn't be seeing this."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "shovel"
+
+/obj/item/weapon/shovel/smithed_shovel/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/shovel_head/S = locate() in contents
+	if(S)
+		var/image/Q = image('icons/obj/blacksmithing.dmi', "shovel_head")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mining/smelter.dm [hunk 38]
```diff
@@ -0,0 +1,331 @@
...
+	desc = "A dwarven buckler."
+	icon_state = "dwarf_buckler"
+	item_state = "dwarf_buckler"
+	block_chance = 30
+
+/obj/item/weapon/shield/riot/buckler/smith/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/shield_backing/S = locate() in contents
+	if(S)
+		var/image/Q = image('icons/obj/blacksmithing.dmi', "dwarf_shield")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mining/smelter.dm [hunk 39]
```diff
@@ -0,0 +1,331 @@
...
+	icon_state = "dwarf_crossbow"
+	item_state = "crossbow"
+	w_class = WEIGHT_CLASS_BULKY
+	mag_type = /obj/item/ammo_box/magazine/internal/speargun/crossbow
+
+/obj/item/weapon/gun/ballistic/automatic/speargun/crossbow/CheckParts(list/parts_list)
+	..()
+	var/obj/item/weapon/mold_result/crossbow_base/S = locate() in contents
+	if(S)
+		var/image/Q = image('icons/obj/blacksmithing.dmi', "dwarf_crossbow")
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: mutable_appearance

File: code/modules/mob/living/carbon/human/interactive.dm [hunk 40]
```diff
@@ -83,7 +83,7 @@
 	var/traitorType = 0
 
 	var/voice_saved = FALSE
-
+	var/allow_random = 1
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/interactive.dm]: TRUE

File: code/modules/mob/living/carbon/human/interactive.dm [hunk 41]
```diff
@@ -1606,6 +1606,19 @@
 	graytide = 1
 	..()
 
+
+/mob/living/carbon/human/interactive/greytide/clown/Initialize()
+	age = rand(AGE_MIN,AGE_MAX)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/interactive.dm]: . = ..() first

File: code/modules/mob/living/carbon/human/interactive.dm [hunk 42]
```diff
@@ -1606,6 +1606,19 @@
...
+	myjob = new/datum/job/clown()
+	job = myjob.title
+	myjob.equip(src)
+	allow_random = 0
+	..()
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/interactive.dm]: Not at the end

File: code/modules/mob/living/carbon/human/interactive.dm [hunk 43]
```diff
@@ -1606,6 +1606,19 @@
 	graytide = 1
 	..()
 
+
+/mob/living/carbon/human/interactive/greytide/clown/Initialize()
+	age = rand(AGE_MIN,AGE_MAX)
+	myjob = new/datum/job/clown()
+	job = myjob.title
+	myjob.equip(src)
+	allow_random = 0
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/interactive.dm]: FALSE

File: code/modules/mob/living/carbon/human/species_types/dwarf.dm [hunk 44]
```diff
@@ -0,0 +1,37 @@
...
+	. = ..()
+	var/dwarf_hair = pick("Dwarf Beard", "Very Long Beard", "Full Beard")
+	var/mob/living/carbon/human/H = C
+	H.grant_language(/datum/language/dwarvish)
+	H.remove_language(/datum/language/common)
+	H.facial_hair_style = dwarf_hair
+	H.update_hair()
+
+	var/obj/item/organ/alcoholvessel/dwarf
+	dwarf = H.getorganslot("dwarf_organ")
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/dwarf.dm]: make a DWARF_ORGAN define please, all organs need to do this

File: code/modules/mob/living/carbon/human/species_types/dwarf.dm [hunk 45]
```diff
@@ -0,0 +1,37 @@
...
+	if(!dwarf)
+		dwarf = new()
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/species_types/dwarf.dm]: remove ()

File: code/modules/projectiles/ammunition/caseless.dm [hunk 46]
```diff
@@ -37,6 +37,24 @@
 	throwforce = 15 //still deadly when thrown
 	throw_speed = 3
 
+/obj/item/crossbow_bolt_spawner/Initialize()
+	..()
+	for(var/i in 1 to 5)
+		new /obj/item/ammo_casing/caseless/crossbow(get_turf(src))
+	qdel(src)
```
Comment (Cyberboss) [inline code/modules/projectiles/ammunition/caseless.dm]: return INITIALIZE_HINT_QDEL

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 47]
```diff
@@ -71,6 +71,39 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	color = "#664300" // rgb: 102, 67, 0
+	nutriment_factor = 1 * REAGENTS_METABOLISM
+	boozepwr = 0 // filled out by brewing
+	taste_description = "ale"
+	glass_name = "glass of ale"
+	glass_desc = "A glass of ale. You feel a fey mood coming on just looking at it."
+	var/datum/reagents/contained_reagents
+
+/datum/reagent/consumable/ethanol/customizable/New()
+	. ..()
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: mmm

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 48]
```diff
@@ -844,13 +858,46 @@
...
+	sharp_result = TRUE
+	durability_reduction = 4 // strike the earth
+
 /datum/reagent/uranium
 	name ="Uranium"
 	id = "uranium"
 	description = "A silvery-white metallic chemical element in the actinide series, weakly radioactive."
 	reagent_state = SOLID
-	color = "#B8B8C0" // rgb: 184, 184, 192
+	color = "#00FF00" // rgb: 184, 184, 192
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: fix the comment

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 49]
```diff
@@ -79,14 +79,19 @@
...
 /datum/chemical_reaction/soapification/on_reaction(datum/reagents/holder, created_volume)
 	var/location = get_turf(holder.my_atom)
 	for(var/i = 1, i <= created_volume, i++)
 		new /obj/item/weapon/soap/homemade(location)
 
+/datum/chemical_reaction/soapification/two
+	name = "Soapification2"
+	id = "soapification2"
+	required_reagents = list("ash" = 10, "water"  = 10)
+	mob_react = 1
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/others.dm]: TRUE

File: code/modules/reagents/reagent_containers/glass.dm [hunk 50]
```diff
@@ -410,4 +425,77 @@
...
+
+/obj/item/weapon/reagent_containers/glass/mold
+	name = "mold"
+	desc = "A clay mold."
+	icon = 'icons/obj/blacksmithing.dmi'
+	icon_state = "mold1"
+	volume = 25
+	amount_per_transfer_from_this = 25
+	flags = OPENCONTAINER
+	var/obj/produce_type = null
```
Comment (Cyberboss) [inline code/modules/reagents/reagent_containers/glass.dm]: remove  ` = null`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mining/smelter.dm [hunk 51]
```diff
@@ -30,21 +30,22 @@
...
 	if(!istype(W, /obj/item/weapon/smith_hammer))
-		return ..()
+		..()
 	if(!current_mold && istype(W, /obj/item/weapon/reagent_containers/glass/mold))
 		var/obj/item/weapon/reagent_containers/glass/mold/M = W
 		var/datum/reagent/R = M.reagents.get_master_reagent()
 		if(R && R.volume == 25)
 			if(user.drop_item())
 				to_chat(user, "You place [M] on [src].")
+				M.loc = src
```
Comment (Cyberboss) [inline code/modules/mining/smelter.dm]: M.forceMove(src)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Tacolizard** [comment]
1. mandates are dumb, because you have to depend on the dwarves to read them. instead the king should be allowed to send messages directly to the dwarves.
2. The king doesn't really have much to do. it's a pretty boring role. it might be more interesting if you had a better method of communication with your dwarves.

**Comment by Tacolizard** [comment]
there also needs to be more weapons, and shields would be cool as well.

**Comment by Shadowlight213** [comment]
150 hours of testing

**Comment by SgtxRootbeer** [comment]
capture

**Comment by Iamgoofball** [comment]
# i pushed a fix the second i figured out what the issue was but we're 30 minutes in and it still hasn't been updated

# blame headmins + ma44

File: code/modules/mining/lavaland/ruins/dwarfruin.dm (hunk 26 above)
Comment (Iamgoofball) [inline code/modules/mining/lavaland/ruins/dwarfruin.dm]: Not changing this, every other role does it.

File: code/game/turfs/simulated/minerals.dm (hunk 14 above)
Comment (Iamgoofball) [inline code/game/turfs/simulated/minerals.dm]: No.

File: code/_onclick/item_attack.dm (hunk 11 above)
Comment (Iamgoofball) [inline code/_onclick/item_attack.dm]: No it doesn't? It actually makes obj_damage work for non-clothing items.

Comment (Cyberboss) [inline code/_onclick/item_attack.dm]: obj_break() != obj_destruction()

Comment (Iamgoofball) [inline code/_onclick/item_attack.dm]: Then what's the point of obj_break()?

Comment (Cyberboss) [inline code/_onclick/item_attack.dm]: for when machines break/clothes tear/etc but aren't destroyed

**Comment by TGxSingularity** [comment]
Dwarf golems should be able to speak dwarf language, kinda hard to obey your master if you can't understand what he's saying. Also, special effect for bananium swords, since dwarves can obtain them from clown portal spawns and ruins?

**Comment by Shadowlight213** [comment]
fix was pushed

**Comment by Grazyn** [comment]
Can't read the code, do mandates still stun you and make you drop your stuff? Because it was the most annoying thing ever

**Comment by Iamgoofball** [comment]
i axed the mandate stun I think

**Comment by Kromgar** [comment]
Oh thank god finally

**Comment by DoctoreMad** [comment]
Dwarves aren't globally told the new mandate anymore. Fuck.

File: code/game/turfs/simulated/minerals.dm (hunk 14 above)
Comment (DoctoreMad) [inline code/game/turfs/simulated/minerals.dm]: doesn't the double // uncomment it

Comment (GunHog) [inline code/game/turfs/simulated/minerals.dm]: Nope, it is entirely safe to do this.

**Comment by Ergovisavi** [comment]
I do have a minor request, if you're going to add new mob types (Those faithless) to lava land, please make them large mobs so the crusher works on them. Thanks.

**Comment by Cyberboss** [comment]
@Iamgoofball

**Comment by Iamgoofball** [comment]
will do @Ergovisavi 

@Cyberboss This is going to depend on the GoofAI PR soon as I'll be integrating some DF mobs using it when it's done.

**Comment by WaifuTed** [comment]
Hope this gets added soon! One of my favourite additions

**Comment by PKPenguin321** [comment]
@Iamgoofball you should finish this and then implement goof AI later tbh, it'd be a shame to see this get hung up even longer when so much good work has gone into it

**Comment by octareenroon91** [comment]
>@Iamgoofball you should finish this and then implement goof AI later tbh, it'd be a shame to see this get hung up even longer when so much good work has gone into it

Dorf code is ambitious. Just get it mergeable please.

**Comment by DoctoreMad** [comment]
>This is going to depend on the GoofAI PR

**Comment by Cyberboss** [comment]
Do the code fixes, nerd

**Comment by SheepFighter** [comment]
>I'll be integrating some DF mobs using it when it's done.

Animal-men Lavaland roles when?

**Comment by Iamgoofball** [comment]
@Cyberboss I need to get GOAP implemented before I can do this, because I need to add Tantrum Spirals.

**Comment by octareenroon91** [comment]
Player-controlled dwarves are sufficiently psychotic. 

# Get this mergeable before you betray MMMiracles' vision

**Comment by GunHog** [comment]
# MERGE THIS AND ADD TO IT LATER

**Comment by Cyberboss** [comment]
Stale

**Comment by PKPenguin321** [comment]
Oof

**Comment by fluxcapacitor1337** [comment]
need this added as a playable race

**Comment by octareenroon91** [comment]
👌

**Pull request closed without merging**